legal-api/scripts/manual_db_scripts/legal_name_change/legal_name_updates.sql

```diff
@@ -1594,8 +1594,53 @@ WHERE d.legal_entity_id = le.id
   AND le.entity_type != 'GP';
 
 
--- TODO offices/offices_history addresses can linked to alternate_names when alternate_name_id has been added to
---  offices table.  Businesses addresses should already be linked for SP persons
+UPDATE offices o
+SET alternate_name_id = an.id,
+    legal_entity_id   = NULL
+FROM alternate_names an
+         JOIN legal_entities le ON an.legal_entity_id = le.id
+WHERE o.legal_entity_id = le.id
+  AND le.entity_type != 'GP';
+
+UPDATE offices_history oh
+SET alternate_name_id = an.id,
+    legal_entity_id   = NULL
+FROM alternate_names an
+         JOIN legal_entities le ON an.legal_entity_id = le.id
+WHERE oh.legal_entity_id = le.id
+  AND le.entity_type != 'GP';
+
+
+-- Link alternate_name_id to custodial address in addresses/addresses_history
+WITH subquery AS (
+    SELECT
+        a.id AS address_id,
+        o.alternate_name_id
+    FROM addresses a
+    JOIN offices o 
+        ON a.office_id = o.id AND COALESCE(a.change_filing_id, 1) = COALESCE(o.change_filing_id, 1) 
```

Comment:
Not sure if it's necessary to check `change_filing_id` in this scenario. If so, I find that some matched address-office records with `change_filing_id` as `NULL` will be excluded. Maybe it's just orphan data but I add `COALESCE()` to prevent omission.

Reply:
Don't believe we need to look at `change_filing_id`.  Can remove this as a join condition